In [1]:
from torch_geometric.datasets import Planetoid

In [2]:
dataset = Planetoid(root='./data/Cora', name='Cora')

Processing...
Done!


In [10]:
print(f'Size                : {len(dataset)}')
print(f'# of classes        : {dataset.num_classes}')
print(f'# of node features  : {dataset.num_node_features}')
print(f'# of edge attributes: {dataset.num_edge_features}')
dataset.print_summary()

Size                : 1
# of classes        : 7
# of node features  : 1433
# of edge attributes: 0
Planetoid (#graphs=1):
+------------+----------+----------+
|            |   #nodes |   #edges |
|------------+----------+----------|
| mean       |     2708 |    10556 |
| std        |      nan |      nan |
| min        |     2708 |    10556 |
| quantile25 |     2708 |    10556 |
| median     |     2708 |    10556 |
| quantile75 |     2708 |    10556 |
| max        |     2708 |    10556 |
+------------+----------+----------+


1433

## CORA dataset

Nodes: 2708 scientific publications
Features: 1433 unique words in the abstract
Edges: Citations

Classes:
1. Case_Based
1. Genetic_Algorithms
1. Neural_Networks
1. Probabilistic_Methods
1. Reinforcement_Learning
1. Rule_Learning
1. Theory

In [11]:
# Cora is 1 single graph
data = dataset[0]

In [12]:
print(f'# of nodes: {data.num_nodes}')
print(f'# of edges: {data.num_edges}')
print(f'# of edge feature: {data.num_edge_features}')
print(f'Has isolated nodes: {data.has_isolated_nodes()}')
print(f'Has self loops: {data.has_self_loops()}') # E.g., A-A is a self-loop
print(f'Is directed: {data.is_directed()}')

# of nodes: 2708
# of edges: 10556
# of edge feature: 0
Has isolated nodes: False
Has self loops: False
Is directed: False


In [15]:
# There are built-in split for train-val-test sets:
print(f'Train:      {data.train_mask.sum().item()}')
print(f'Validation: {data.val_mask.sum().item()}')
print(f'Test:       {data.test_mask.sum().item()}')

Train:      140
Validation: 500
Test:       1000
